obspy/core/trace.py

```diff
@@ -2779,6 +3241,207 @@ def _data_sanity_checks(value):
                "arrays are allowed for initialization.") % str(value.shape)
         raise ValueError(msg)
 
+class FrequencyDomainTrace(BaseTrace):
+
+    def ifft(self):
+        tr = Trace()
+        tr.stats = Stats()
+        tr.stats.starttime = self.stats.starttime
+        tr.data = fftpack.irfft(self.data)
```

Comment:
> extremely awkward return type

IIRC that return type is for historic reasons, I think it was like this in old numerical recipes C codes. There might be a chance to convert it without copy when leaving out the zero frequency value, and prepend that one afterwards?